cli/src/alluxio.org/cli/processes/common.go

```diff
@@ -0,0 +1,172 @@
+
+	"golang.org/x/crypto/ssh"
+
+	"alluxio.org/cli/env"
+	"alluxio.org/log"
+)
+
+func getMasters() ([]string, error) {
+	mastersDir := path.Join(env.Env.EnvVar.GetString(env.ConfAlluxioConfDir.EnvVar), "masters")
+	mastersFile, err := os.Open(mastersDir)
+	if err != nil {
+		log.Logger.Errorf("Error reading worker hostnames at %s", mastersDir)
+		return nil, err
+	}
+
+	mastersReader := bufio.NewReader(mastersFile)
+	var mastersList []string
+	lastLine := false
+	for !lastLine {
+		// read lines of the workers file
+		line, err := mastersReader.ReadString('\n')
+		if err != nil {
+			if err == io.EOF {
+				lastLine = true
+			} else {
+				log.Logger.Errorf("Error parsing worker file at this line: %s", line)
+				return nil, err
+			}
+		}
+		// remove notes
+		if strings.Index(line, "#") != -1 {
+			line = line[:strings.Index(line, "#")]
+		}
+		line = strings.TrimSpace(line)
+		if line != "" {
+			mastersList = append(mastersList, line)
+		}
+	}
+	return mastersList, nil
+}
+
+func getWorkers() ([]string, error) {
+	workersDir := path.Join(env.Env.EnvVar.GetString(env.ConfAlluxioConfDir.EnvVar), "workers")
+	workersFile, err := os.Open(workersDir)
+	if err != nil {
+		log.Logger.Errorf("Error reading worker hostnames at %s", workersDir)
+		return nil, err
+	}
+
+	workersReader := bufio.NewReader(workersFile)
+	var workersList []string
+	lastLine := false
+	for !lastLine {
+		// read lines of the workers file
+		line, err := workersReader.ReadString('\n')
+		if err != nil {
+			if err == io.EOF {
+				lastLine = true
+			} else {
+				log.Logger.Errorf("Error parsing worker file at this line: %s", line)
+				return nil, err
+			}
+		}
+		// remove notes
+		if strings.Index(line, "#") != -1 {
+			line = line[:strings.Index(line, "#")]
+		}
+		line = strings.TrimSpace(line)
+		if line != "" {
+			workersList = append(workersList, line)
+		}
+	}
+	return workersList, nil
+}
+
+func getPrivateKey() (ssh.Signer, error) {
+	homePath, err := os.UserHomeDir()
+	if err != nil {
+		log.Logger.Errorf("User home directory not found at %s", homePath)
+		return nil, err
+	}
+	privateKey, err := os.ReadFile(path.Join(homePath, ".ssh", "id_rsa"))
+	if err != nil {
+		log.Logger.Errorf("Private key file not found at %s", path.Join(homePath, ".ssh", "id_rsa"))
+		return nil, err
+	}
+	parsedPrivateKey, err := ssh.ParsePrivateKey(privateKey)
+	if err != nil {
+		log.Logger.Errorf("Cannot parse public key at %s", path.Join(homePath, ".ssh", "id_rsa"))
+		return nil, err
+	}
+	return parsedPrivateKey, nil
+}
+
+func dialConnection(remoteAddress string, key ssh.Signer) (*ssh.Client, error) {
+	clientConfig := &ssh.ClientConfig{
+		// TODO: how to get user name? Like ${USER} in alluxio-common.sh
+		User: "root",
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(key),
+		},
+		Timeout:         5 * time.Second,
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	// TODO: Some machines might have changed default SSH port. Get ssh port or remind users when get started.
```

Comment:
similar to the private key, this could be configurable with a new flag for the command